games-emulation/retroarch/retroarch_assets-20180129.recipe

```diff
@@ -0,0 +1,28 @@
+SUMMARY="Media assets for RetroArch"
+DESCRIPTION="RetroArch is a frontend for emulators, game engines and media \
+players. This package provides assets such as icons for the GUI."
+HOMEPAGE="https://libretro.com/"
+COPYRIGHT="2010-2018 The RetroArch Team"
+LICENSE="GNU GPL v3"
+REVISION="1"
+srcGitRev="505ad6205e5fd8a155daad02ab39953a0b15b0a9"
+SOURCE_URI="https://github.com/libretro/retroarch-assets/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="de37772b00b755b80aafe5d4df0713c0fe6b0e5a79017d94e43f279e798aa85d"
+SOURCE_FILENAME="retroarch_assets-$portVersion.tar.gz"
+SOURCE_DIR="retroarch-assets-$srcGitRev"
+
+ARCHITECTURES="any"
+
+PROVIDES="
+	retroarch_assets = $portVersion
+	"
+
+INSTALL()
+{
+	retroarchDataDir="$dataDir/retroarch/assets/"
+	mkdir -p "$retroarchDataDir"
+	cp -ar * "$retroarchDataDir"
+	rm -rf "$retroarchDataDir"/Makefile \
```

Comment:
Since the tarball is quite big (around 86 MiB), I'd like to suggest a small variant, but I don't know if it is acceptable because it does something (`rm ../../flag.unpack ../../flag.build`) which violates an implicit rule. This variant works like a charm, but we need to wait for the green lights from at least a few key contributors.

```diff
 PROVIDES="
 	retroarch_assets = $portVersion
 	"
+REQUIRES="
+	haiku
+	"

 INSTALL()
 {
 	retroarchDataDir="$dataDir/retroarch/assets/"
 	mkdir -p "$retroarchDataDir"
-	cp -ar * "$retroarchDataDir"
-	rm -rf "$retroarchDataDir"/Makefile \
-		"$retroarchDataDir"/configure \
-		"$retroarchDataDir"/src
+	rm ../../flag.unpack ../../flag.build Makefile configure
+	rm -rf src
+	mv * "$retroarchDataDir"
 }
```

If this variant is not acceptable, then the only changes I see are:
* adding a `BUILD()` that would `rm configure Makefile` and `rm -rf src`.
* adjusting `INSTALL()` accordingly.
* Adding `REQUIRES` as suggested above.
* addressing the `LICENSE` topic... which does not currently mention the real licenses used by the assets.

Reply:
The actual license is CC-BY-4.0 according to github, which is not in the list of included licenses :(

````
Error: Error: No match found for license CC-BY-4.0
Error: Error: Valid license filenames included with Haiku are:
Error: Error: Anti-Grain Geometry, Apache v2, Artistic, Artistic v2.0, BSD (2-clause), BSD (3-clause), BSD (4-clause), Be Sample Code License, Bitstream Charter, CDDL v1, CQuantizer, Curl, DEC, Flex, FreeType, GNU GPL font exception, GNU GPL v1, GNU GPL v2, GNU GPL v3, GNU LGPL v2, GNU LGPL v2.1, GNU LGPL v3, ICU, Info-ZIP, Intel (2xxx firmware), Intel (ACPICA), Intel (firmware), JPEG, Jam, JasPer v2, LibJPEG, LibPNG, MAPM, MIT, MIT (no promotion), MPL v1.1, MPL v2.0, Marvell (firmware), OpenGroup, OpenSSH, OpenSSL, PCRE, Public Domain, Python, Ralink (firmware), SGI Free B, SIL Open Font License v1.1, SQLite, Speex, UIUC, Vim, WebKit Apple, Zlib, bzip2
````

Reply:
According to [this page](https://wiki.creativecommons.org/wiki/ShareAlike_compatibility) CC-BY-4.0 is one-way compatible with the GNU GPL v3 (see the bottom of the page) so we should be fine with the GPL v3.

Reply:
I kept CC-BY-4.0 commented out to reflect this in my latest commit.

Reply:
Yep, and that's not funny...
Whenever a recipe has one or more licenses that are not included in `haiku-r1~alpha4_pm_hrev*-*-*.hpkg`, the missing licenses need to be added to the recipe's `licenses` subdirectory.
So, for retroarch_assets, we need to put in `haikuports/games-emulation/retroarch/licenses/` a copy of the COPYING file (from the retroarch-asset tarball) renamed to CC-BY-4.0.
BTW, since we'll have `LICENSE="CC-BY-4.0"`, the license will be available  in `/boot/system/data/licenses/` (when the package is installed) so we can remove the COPYING file before the installation to `$dataDir/retroarch/` (just like we removed the Makefile and configure files.)

But if, in the future, the `CC-BY-4.0` file is added to haiku, then it will be OK to remove it from `haikuports/games-emulation/retroarch/licenses/`.
However, this also means that the package would no longer contain a copy of the license, so, after all, it might be better to keep the COPYING file.
(BTW, we really need to decide what to do with all these popular licenses (e.g. GPL), because we are not embedding them in the package.)

Reply:
Ok I've made a new push following these recommendations